okhttp/src/main/java/okhttp3/Handshake.kt

```diff
@@ -152,20 +155,20 @@ class Handshake private constructor(
         null
       }
 
-      val peerCertificatesList = if (peerCertificates != null) {
-        immutableListOf(*peerCertificates)
-      } else {
-        emptyList()
-      }
-
       val localCertificates = localCertificates
       val localCertificatesList = if (localCertificates != null) {
         immutableListOf(*localCertificates)
       } else {
         emptyList()
       }
 
-      return Handshake(tlsVersion, cipherSuite, peerCertificatesList, localCertificatesList)
+      return Handshake(tlsVersion, cipherSuite, localCertificatesList) {
+        if (peerCertificates != null) {
```

Comment:
I think it’d be simpler to continue to compute the `peerCertificatesList` eagerly and just make this a boring function. Deferring the immutableListOf() call is more complexity than necessary!